android-app/src/main/java/org/example/app/di/UnitFactoryModule.kt

```diff
@@ -0,0 +1,16 @@
+package org.example.app.di
+
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+import org.example.app.units.NewsListUnitsFactory
+import org.example.library.SharedFactory
+
+@InstallIn(SingletonComponent::class)
```

Comment:
why it's singleton? units factory should be activity-scoped i think

Reply:
Yes, we could do this, but right now `SharedFactory` needs `NewsUnitFactory`'s implementation. So if we want it to be Activity Scoped, we have to refactor current code to provide unit factory from activity.